controllers/tests/controller_server/host_definer/csi_node_watcher_test.py

```diff
@@ -0,0 +1,322 @@
+                      settings.FAKE_NODE_NAME), self._mock_logger.records)
+
+    def test_failed_update_node_label(self):
+        self.csi_node_watcher.csi_nodes_api.get.return_value = self.fake_k8s_csi_nodes_with_ibm_block
+        self.mock_os.getenv.return_value = settings.TRUE_STRING
+        self.csi_node_watcher.core_api.read_node.return_value = self.fake_k8s_node_with_fake_label
+        self.csi_node_watcher.core_api.patch_node.side_effect = self.fake_api_exception
+        self.csi_node_watcher.add_initial_csi_nodes()
+        utils.assert_fail_to_update_label_log_message(self.http_resp.data, self._mock_logger.records)
+        self.assertEqual(1, len(self.mock_nodes_on_watcher_helper))
+
+    def test_failed_to_get_csi_nodes(self):
+        self.csi_node_watcher.csi_nodes_api.get.side_effect = self.fake_api_exception
+        self.csi_node_watcher.add_initial_csi_nodes()
+        self.assertIn(messages.FAILED_TO_GET_CSI_NODES.format(self.http_resp.data), self._mock_logger.records)
+
+
+class TestWatchCsiNodesResources(BaseSetUp):
+    def setUp(self):
+        super().setUp()
+        self.csi_node_watcher._get_k8s_object_resource_version = Mock()
+        self.csi_node_watcher._get_k8s_object_resource_version.return_value = settings.FAKE_RESOURCE_VERSION
+        self.csi_node_watcher._delete_host_definition = Mock()
+
+    def test_do_nothing_when_csi_node_is_updated(self):
+        self.csi_node_watcher.csi_nodes_api.watch.return_value = iter(
+            [utils.get_fake_csi_node_watch_event(settings.DELETED_EVENT_TYPE)])
+        self.csi_node_watcher.core_api.read_node.return_value = self.fake_k8s_node_with_manage_node_label
+        self.csi_node_watcher.apps_api.list_daemon_set_for_all_namespaces.side_effect = [
+            self.fake_not_updated_daemon_set, self.fake_updated_daemon_set]
+        self.csi_node_watcher.core_api.list_pod_for_all_namespaces.return_value = self.fake_k8s_pods
+        utils.run_function_with_timeout(self.csi_node_watcher.watch_csi_nodes_resources, 0.5)
+        self.assertEqual(1, len(self.mock_nodes_on_csi_node_watcher))
+
+    def test_delete_host_definition(self):
+        self._default_mock_csi_node_pod_deletion()
+        utils.run_function_with_timeout(self.csi_node_watcher.watch_csi_nodes_resources, 0.5)
+        self.csi_node_watcher._delete_host_definition.assert_called()
+
+    def test_delete_host_definition_undefine_host_log_message(self):
+        self._default_mock_csi_node_pod_deletion()
+        utils.run_function_with_timeout(self.csi_node_watcher.watch_csi_nodes_resources, 0.5)
+        self.assertIn(messages.UNDEFINED_HOST.format(settings.FAKE_NODE_NAME,
+                      settings.FAKE_SECRET, settings.FAKE_SECRET_NAMESPACE), self._mock_logger.records)
+
+    def test_nodes_global_variable_reduced_on_csi_node_deletion(self):
+        self._default_mock_csi_node_pod_deletion()
+        utils.run_function_with_timeout(self.csi_node_watcher.watch_csi_nodes_resources, 0.5)
```

Comment:
redundant, can add to [here ](https://github.com/IBM/ibm-block-csi-driver/pull/579/files#diff-3780fbb9b4124cf1e64f2a42f147dc1cd5fc09cdff8e12ea34300054a182012eR103)

Reply:
I am not sure what you mean

Reply:
the same test with different assert at the end.. can be unified..

Reply:
I think that there is no `test_nodes_global_variable_reduced_on_csi_node_deletion` test anymore